src/widgets/kernel/qwidget.cpp

```diff
@@ -6389,23 +6389,48 @@ void QWidget::setFocusProxy(QWidget * w)
         // the focus chain.
         const QWidget *parentOfW = w->parentWidget();
         Q_ASSERT(parentOfW); // can't be nullptr since we are an ancestor of w
+
```

Comment:
Ok this bit was horrible. I _think_ what I have is better...

Previously it basically broke the focus order by moving the parent widget to directly behind its first child in the focus chain - even if its first child in the children list is not the first child in the focus chain. This meant that children before that one in the chain just end up in a random position in the tab order, which of course broke guitests